molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java

```diff
@@ -391,6 +399,31 @@ private boolean isGenomeBrowserEntity(EntityMetaData entityMetaData)
 		return attributeStartPosition != null && attributeChromosome != null;
 	}
 
+	/**
+	 * Updates the 'Entities' menu when an entity is deleted.
+	 * 
+	 * @param datasetName
+	 */
+	@RequestMapping(value = "/removeEntityFromMenu/{entityName}", method = POST)
+	public void updateMenuOnDeleteEntity(@PathVariable("entityName") String entityName, HttpServletResponse response)
+	{
+		Menu menu = menuManager.getMenu();
+		MenuItem mEntities = menu.findMenuItem("entities");
+
+		List<MenuItem> items = mEntities.getItems();
+		List<MenuItem> newItems = Lists.newArrayList();
+		for (MenuItem i : items)
+		{
+			if (!i.getLabel().equals(entityName))
```

Comment:
User might rename the plugin thus changing the label, check for id instead of label.